kotlin/add-features-with-contingent-values/src/main/java/com/esri/arcgisruntime/sample/addfeatureswithcontingentvalues/MainActivity.kt

```diff
@@ -0,0 +1,447 @@
+    }
+
+    /**
+     *  Retrieve the buffer attribute fields, add the fields to a ContingentRangeValue,
+     *  and set the values to a SeekBar
+     */
+    private fun showBufferSeekbar() {
+        // set the bottom sheet view to enable the buffer attribute
+        bottomSheetBinding.apply {
+            bufferSeekBar.isEnabled = true
+            selectedBuffer.text = ""
+        }
+
+        // get the contingent value results using the feature and field
+        val contingentValueResult = featureTable.getContingentValues(feature, "BufferSize")
+        val bufferSizeGroupContingentValues =
+            (contingentValueResult.contingentValuesByFieldGroup["BufferSizeFieldGroup"]?.get(0) as ContingentRangeValue)
+        // set the minimum and maximum possible buffer sizes
+        val minValue = bufferSizeGroupContingentValues.minValue as Int
+        val maxValue = bufferSizeGroupContingentValues.maxValue as Int
+        // check if there can be a max value, if not disable SeekBar & set value to attribute size to 0
+        if (maxValue > 0) {
+            // get SeekBar instance from the binding layout
+            val bufferSeekBar = bottomSheetBinding.bufferSeekBar
+            // set the min, max and current value of the SeekBar
+            bufferSeekBar.valueFrom = minValue.toFloat()
+            bufferSeekBar.valueTo = maxValue.toFloat()
+            bufferSeekBar.value = bufferSeekBar.valueFrom
+            // set the change listener to update the attribute value and the displayed value to the SeekBar position
+            bufferSeekBar.addOnChangeListener { _, value, _ ->
+                feature.attributes["BufferSize"] = value.toInt()
+                bottomSheetBinding.selectedBuffer.text = value.toInt().toString()
+            }
+        } else {
+            // max value is 0, so disable seekbar and update the attribute value accordingly
+            bottomSheetBinding.apply {
+                bufferSeekBar.isEnabled = false
+                selectedBuffer.text = "0"
+            }
+            feature.attributes["BufferSize"] = 0
+        }
+    }
+
+    /**
+     * Set up the [feature] using the status attribute's coded value
+     * by loading the [featureTable]'s Contingent Value Definition.
+     * This function calls setUpProtectionAttributes() once the [feature] has been set
+     */
+    private fun createFeature(codedValue: CodedValue) {
+        // get the contingent values definition from the feature table
+        val contingentValueDefinition = featureTable.contingentValuesDefinition
+        // load the contingent values definition
+        contingentValueDefinition.loadAsync()
+        contingentValueDefinition.addDoneLoadingListener {
+            // create a feature from the feature table and set the initial attribute
+            feature = featureTable.createFeature() as ArcGISFeature
+            feature.attributes["Status"] = codedValue.code
+            setUpProtectionAttributes()
+        }
+    }
```

Comment:
maybe this method should go before setupProtectionAttributes, so its closer to the order the codes being run in

Reply:
Not sure where you are referring to placing this method. Currently here, `setUpProtectionAttributes()` needs to be called after the feature attribute has been set to the status-coded value.

Reply:
From what I can tell `setupProtectionAttributes()` is only called by one method `createFeature()`. So if we consider the code being read top to bottom, it feels more intuitive to me that `setupProtectionAttributes()` should come after `createFeature()`

Reply:
Oh, you meant from a readability perspective. I understand now, and made the change :)